ast-nodes/if-statement.dart

```diff
@@ -12,6 +12,32 @@ class IfStatement implements Statement {
 
   // TODO: implement .parse()
 
+  factory IfStatement.parse(Iterable<Token> tokens) {
+    var iterator = tokens.iterator;
+    checkNext(iterator, RegExp('if\$'), "Expected 'if'");
+    var expressionBody = consumeUntil(iterator, RegExp("^then\$"));
+    checkNext(iterator, RegExp('then\$'), "Expected 'then'");
+    var body = consumeUntil(iterator, RegExp("^else\$"));
+    checkNext(iterator, RegExp('else\$'), "Expected 'else'");
+    var elseBody = consumeUntil(iterator, RegExp("^end\$"));
+    checkThis(iterator, RegExp('end\$'), "Expected 'end'");
+    checkNoMore(iterator);
+
+    if (expressionBody.isEmpty) {
+      throw SyntaxError(iterator.current, "Expected a condition");
+    }
+
+    if (body.isEmpty) {
+      throw SyntaxError(iterator.current, "Expected a true statement");
+    }
+
+    var expBody = Expression.parse(expressionBody);
+    var trueBlock = List<Statement>.parse(body);
```

Comment:
List has no `parse` method. You should split statements by \n or ; and parse them one by one (see other similar files)